```
/Modules/Core/TestingHelper/src/mitkInteractionTestHelper.cpp:176:79: error: no matching function for call to ‘mitk::MouseModeSwitcher::New(const char [22])’
     m_MouseModeSwitcher = mitk::MouseModeSwitcher::New("InteractionTestHelper");```

--- a/Modules/Core/TestingHelper/src/mitkInteractionTestHelper.cpp
+++ b/Modules/Core/TestingHelper/src/mitkInteractionTestHelper.cpp
@@ -173,7 +173,11 @@ void mitk::InteractionTestHelper::Initialize(const std::string &interactionXmlFi
 
     //########### register display interactor to handle scroll events ##################
     //use MouseModeSwitcher to ensure that the statemachine of DisplayInteractor is loaded correctly
-    m_MouseModeSwitcher = mitk::MouseModeSwitcher::New("InteractionTestHelper");
+    m_MouseModeSwitcher = mitk::MouseModeSwitcher::New();
+
+    for (const auto& window : m_RenderWindowList) {
+      m_MouseModeSwitcher->AddRenderer(window->GetRenderer());
+    }
   }
   else
   {

--- a/Modules/Core/files.cmake
+++ b/Modules/Core/files.cmake
@@ -297,6 +297,7 @@ set(CPP_FILES
 set(RESOURCE_FILES
 Interactions/globalConfig.xml
 Interactions/DisplayInteraction.xml
+Interactions/DisplayInteraction3D.xml
 Interactions/DisplayConfig.xml
 Interactions/DisplayConfigPanLeft.xml
 Interactions/DisplayConfigPanRight.xml

--- a/Modules/Core/include/mitkDisplayInteractor.h
+++ b/Modules/Core/include/mitkDisplayInteractor.h
@@ -51,6 +51,8 @@ namespace mitk
     void SetSelectionMode(bool selection);
     static bool GetMouseRotationMode();
 
+    void SetOnly3D(bool only3D) { m_Only3D = only3D; }
+
     static void SetSynchronization(bool on, bool studySync);
 
     static void RotateCamera(BaseRenderer* renderer, bool clockwise);
@@ -288,6 +290,9 @@ namespace mitk
      * @brief m_LinkPlanes Determines if angle between crosshair remains fixed when rotating
      */
     bool m_LinkPlanes;
+
+    // If set will be filtering all 2d events out. Otherwise all 3d events
+    bool m_Only3D;
 
     SNCVector m_RotatableSNCs; /// all SNCs that currently have CreatedWorldGeometries, that can be rotated.
     SNCVector m_SNCsToBeRotated; /// all SNCs that will be rotated (exceptions are the ones parallel to the one being clicked)

--- a/Modules/Core/include/mitkMouseModeSwitcher.h
+++ b/Modules/Core/include/mitkMouseModeSwitcher.h
@@ -72,8 +72,8 @@ namespace mitk {
 #pragma GCC visibility pop
 
     mitkClassMacroItkParent( MouseModeSwitcher, itk::Object );
-    mitkNewMacro1Param(Self, const std::string&);
-    itkCloneMacro(Self);
+    mitkNewMacro1Param(Self, BaseRenderer::Pointer);
+    itkNewMacro(Self);
 
     // enum of the different interaction schemes that are available
     enum InteractionScheme
@@ -126,10 +126,11 @@ namespace mitk {
     /**
     * \brief Initializes the listener with the MITK default behavior.
     */
-    void AddRenderer(const std::string& rendererName);
+    void AddRenderer(BaseRenderer::Pointer renderer);
 
   protected:
-    MouseModeSwitcher(const std::string& rendererName = "");
+    MouseModeSwitcher(BaseRenderer::Pointer renderer);
+    MouseModeSwitcher();
     virtual ~MouseModeSwitcher();
   private:
 

--- a/Modules/Core/resource/Interactions/DisplayConfigMITK.xml
+++ b/Modules/Core/resource/Interactions/DisplayConfigMITK.xml
@@ -41,6 +41,9 @@
   <event_variant class="MousePressEvent" name="SetCrosshairL">
     <attribute name="EventButton" value="LeftMouseButton"/>
   </event_variant>
+  <event_variant class="MouseDoubleClickEvent" name="DoubleClick">
+    <attribute name="EventButton" value="LeftMouseButton"/>
+  </event_variant>
 
   <!-- Moving -->
   <event_variant name="Move" class="MouseMoveEvent"/>

--- a/Modules/Core/resource/Interactions/DisplayInteraction3D.xml
+++ b/Modules/Core/resource/Interactions/DisplayInteraction3D.xml
@@ -1,0 +1,7 @@
+<statemachine>
+    <state name="start" startstate="true" >
+        <transition event_class="InteractionPositionEvent" event_variant="DoubleClick" target="start">
+          <action name="doubleClick"/>
+        </transition>
+    </state>
+</statemachine>

--- a/Modules/Core/src/Interactions/mitkDisplayInteractor.cpp
+++ b/Modules/Core/src/Interactions/mitkDisplayInteractor.cpp
@@ -77,7 +77,9 @@ void mitk::DisplayInteractor::ConnectActionsAndFunctions()
   CONNECT_FUNCTION("levelWindow", AdjustLevelWindow);
   CONNECT_FUNCTION("setCrosshair", SetCrosshair);
 
-  CONNECT_FUNCTION ("updateStatusbar", UpdateStatusbar)
+  CONNECT_FUNCTION("doubleClick", SetCrosshair);
+
+  CONNECT_FUNCTION("updateStatusbar", UpdateStatusbar)
   CONNECT_FUNCTION("startRotation", StartRotation);
   CONNECT_FUNCTION("endRotation", EndRotation);
   CONNECT_FUNCTION("rotationModeChanged", EndRotation);
@@ -87,6 +89,7 @@ void mitk::DisplayInteractor::ConnectActionsAndFunctions()
 
   CONNECT_FUNCTION("rotateClock", RotateClock);
   CONNECT_FUNCTION("rotateBackClock", RotateBackClock);
+
   CONNECT_FUNCTION("selectObject", SelectObject);
   CONNECT_FUNCTION("deSelectObject", DeSelectObject);
 
@@ -110,6 +113,7 @@ mitk::DisplayInteractor::DisplayInteractor()
   , m_ZoomFactor(2)
   , m_LinkPlanes(true)
   , m_SelectionMode(false)
+  , m_Only3D(false)
 {
   m_StartCoordinateInMM.Fill(0);
   m_LastDisplayCoordinate.Fill(0);
@@ -1161,10 +1165,15 @@ void mitk::DisplayInteractor::ConfigurationChanged()
 
 bool mitk::DisplayInteractor::FilterEvents(InteractionEvent* interactionEvent, DataNode* /*dataNode*/)
 {
-  if (interactionEvent->GetSender() == nullptr)
+  if (interactionEvent->GetSender() == nullptr) {
     return false;
-  if (interactionEvent->GetSender()->GetMapperID() == BaseRenderer::Standard3D)
+  }
+  if (!m_Only3D && interactionEvent->GetSender()->GetMapperID() == BaseRenderer::Standard3D) {
     return false;
+  }
+  if (m_Only3D && interactionEvent->GetSender()->GetMapperID() == BaseRenderer::Standard2D) {
+    return false;
+  }
 
   return true;
 }

--- a/Modules/Core/src/Interactions/mitkMouseModeSwitcher.cpp
+++ b/Modules/Core/src/Interactions/mitkMouseModeSwitcher.cpp
@@ -56,16 +56,21 @@ namespace {
   }
 }
 
-mitk::MouseModeSwitcher::MouseModeSwitcher(const std::string& rendererName) :
-    m_ActiveInteractionScheme(MITK), m_ActiveMouseMode(MousePointer)
+mitk::MouseModeSwitcher::MouseModeSwitcher() :
+  m_ActiveInteractionScheme(MITK), m_ActiveMouseMode(MousePointer)
 {
   m_ActiveMouseModes.clear();
   m_ActiveMouseModes[mitk::MouseModeSwitcher::CrossHair] = { 0x00000001 }; // Qt::LeftButton
   m_ActiveMouseModes[mitk::MouseModeSwitcher::Zoom] = { 0x00000002 }; // Qt::RightButton
   m_ActiveMouseModes[mitk::MouseModeSwitcher::Pan] = { 0x00000004 }; // Qt::MiddleButton
 
-  this->AddRenderer(rendererName);
   this->SetInteractionScheme(m_ActiveInteractionScheme);
+}
+
+mitk::MouseModeSwitcher::MouseModeSwitcher(mitk::BaseRenderer::Pointer renderer) :
+    mitk::MouseModeSwitcher::MouseModeSwitcher()
+{
+  this->AddRenderer(renderer);
 }
 
 mitk::MouseModeSwitcher::~MouseModeSwitcher()
@@ -75,17 +80,30 @@ mitk::MouseModeSwitcher::~MouseModeSwitcher()
   }
 }
 
-void mitk::MouseModeSwitcher::AddRenderer(const std::string& rendererName)
+void mitk::MouseModeSwitcher::AddRenderer(mitk::BaseRenderer::Pointer renderer)
 {
   DisplayInteractor::Pointer currentObserver = mitk::DisplayInteractor::New();
-  currentObserver->LoadStateMachine("DisplayInteraction.xml");
   currentObserver->SetEventConfig("DisplayConfigMITK.xml");
+
+  if (renderer->GetMapperID() == mitk::BaseRenderer::Standard2D) {
+    currentObserver->LoadStateMachine("DisplayInteraction.xml");
+    currentObserver->SetOnly3D(false);
+  } else if (renderer->GetMapperID() == mitk::BaseRenderer::Standard3D) {
+    currentObserver->LoadStateMachine("DisplayInteraction3D.xml");
+    currentObserver->SetOnly3D(true);
+  } else {
+    assert(false);
+  }
+
   // Register as listener via micro services
   us::ServiceProperties props;
   props["name"] = std::string("DisplayInteractor");
+
+  std::string rendererName = renderer->GetName();
   if (!rendererName.empty()) {
     props["rendererName"] = rendererName;
   }
+
   us::ServiceRegistration<InteractionEventObserver> serviceRegistration = 
     us::GetModuleContext()->RegisterService<InteractionEventObserver>(currentObserver.GetPointer(),props);
 

--- a/Modules/QmlItems/src/QmlMitkStdMultiItem.cpp
+++ b/Modules/QmlItems/src/QmlMitkStdMultiItem.cpp
@@ -61,13 +61,18 @@ void QmlMitkStdMultiItem::init()
     if(QmlMitkStdMultiItem::storage.IsNull())
         QmlMitkStdMultiItem::storage = mitk::StandaloneDataStorage::New();
 
-    this->m_mouseMode = mitk::MouseModeSwitcher::New("QmlMitkStdMultiItem");
+    this->m_mouseMode = mitk::MouseModeSwitcher::New();
     this->m_mouseMode->SetInteractionScheme(mitk::MouseModeSwitcher::InteractionScheme::MITK);
 
     this->m_viewerAxial->SetDataStorage(QmlMitkStdMultiItem::storage);
     this->m_viewerFrontal->SetDataStorage(QmlMitkStdMultiItem::storage);
     this->m_viewerSagittal->SetDataStorage(QmlMitkStdMultiItem::storage);
     this->m_viewerOriginal->SetDataStorage(QmlMitkStdMultiItem::storage);
+
+    m_mouseMode->AddRenderer(m_viewerAxial->GetRenderer());
+    m_mouseMode->AddRenderer(m_viewerFrontal->GetRenderer());
+    m_mouseMode->AddRenderer(m_viewerSagittal->GetRenderer());
+    m_mouseMode->AddRenderer(m_viewerOriginal->GetRenderer());
 
     this->addPlanes();
 

--- a/Modules/QtWidgets/src/QmitkStdMultiWidget.cpp
+++ b/Modules/QtWidgets/src/QmitkStdMultiWidget.cpp
@@ -378,11 +378,11 @@ void QmitkStdMultiWidget::InitializeWidget(bool showPlanesIn3d)
   //mitkWidget4->GetSliceNavigationController()
   //  ->ConnectGeometryTimeEvent(m_TimeNavigationController, false);
 
-  m_MouseModeSwitcher = mitk::MouseModeSwitcher::New(mitkWidget1->GetRenderer()->GetName());
+  m_MouseModeSwitcher = mitk::MouseModeSwitcher::New(mitkWidget1->GetRenderer());
   //m_MouseModeSwitcher->AddRenderer(mitkWidget1->GetRenderer()->GetName());
-  m_MouseModeSwitcher->AddRenderer(mitkWidget2->GetRenderer()->GetName());
-  m_MouseModeSwitcher->AddRenderer(mitkWidget3->GetRenderer()->GetName());
-  m_MouseModeSwitcher->AddRenderer(mitkWidget4->GetRenderer()->GetName());
+  m_MouseModeSwitcher->AddRenderer(mitkWidget2->GetRenderer());
+  m_MouseModeSwitcher->AddRenderer(mitkWidget3->GetRenderer());
+  m_MouseModeSwitcher->AddRenderer(mitkWidget4->GetRenderer());
 
   mitkWidget1->GetSliceNavigationController()->crosshairPositionEvent.AddListener(mitk::MessageDelegate<QmitkStdMultiWidget>(this, &QmitkStdMultiWidget::HandleCrosshairPositionEvent));
   mitkWidget2->GetSliceNavigationController()->crosshairPositionEvent.AddListener(mitk::MessageDelegate<QmitkStdMultiWidget>(this, &QmitkStdMultiWidget::HandleCrosshairPositionEvent));

